src/config.cpp

```diff
@@ -38,13 +39,17 @@ void EmulatorConfig::load(const std::filesystem::path& path) {
 void EmulatorConfig::save(const std::filesystem::path& path) {
 	toml::basic_value<toml::preserve_comments> data;
 
-	if (std::filesystem::exists(path)) {
+	std::error_code error;
+	if (std::filesystem::exists(path, error)) {
 		try {
 			data = toml::parse<toml::preserve_comments>(path);
 		} catch (std::exception& ex) {
 			Helpers::warn("Got exception trying to save config file. Exception: %s\n", ex.what());
 			return;
 		}
+	} else {
+		Helpers::warn("Filesystem error accessing %s: Exception: %s\n", path.string().c_str(), error.message().c_str());
```

Comment:
Hmm, if this code path is reached it doesn't mean there was necessarily an error, does it? Only that the file doesn't exist which I don't think qualifies as an error?

I think it should be like

```cc
Helpers::warn("Filesystem error accessing %s", path.string().c_str());
if (error) {
    Helpers::warn("Exception: %s", ...);
}
```

Reply:
Other than that it's good, tell me if you want to fix it or if you prefer me doing it instead on Discord

Reply:
I can fix it, it def shouldn't be an error if the file just doesn't exists. I'll have it print out an error code if there was an error while checking if the file exists or not, and print out that it is creating a new configuration file in the case that it doesn't.